include/crm/crm.h

```diff
@@ -51,7 +51,7 @@ extern "C" {
  * >=3.0.13: Fail counts include operation name and interval
  * >=3.2.0:  DC supports PCMK_LRM_OP_INVALID and PCMK_LRM_OP_NOT_CONNECTED
  */
-#  define CRM_FEATURE_SET		"3.3.0"
+#  define CRM_FEATURE_SET		"3.4.0"
```

Comment:
I have this nagging feeling that something, somewhere, needs to be updated every time CRM_FEATURE_SET is changed but I can't think of what it might be.  Sorry that's not very useful, but maybe it'll spark you to remember something else (or that I am making things up).

Reply:
We update some docs at the next release, but nothing really needs to be done for the bump itself. The comment above this documents particular versions of significance to the code -- not every bump involves code checking for that version.

BTW I forgot about this for the rsc_expression/op_expression PR, we really should have bumped it then -- but it doesn't matter as long as it gets bumped (here) before the next release. In other words we don't need to bump for every feature merged into master, as long as the next release includes it. So after this PR is merged, we don't really need to bump feature set again (regardless of what gets added later) until after the next release.

The point of the feature set is for mixed version clusters and rolling upgrades. Without a feature set bump, pacemaker is happy to have a permanent mixed version cluster with different pacemaker versions that all have the same feature set. With a feature set bump, mixed version clusters are allowed only during a rolling upgrade (if a node with the old feature set leaves the cluster, it will not be allowed to rejoin unless it is upgraded).

The last piece of the puzzle is that only a node with the oldest feature set in the cluster can be elected DC. This ensures that new features (in the controller/scheduler) do not take effect until all nodes support them. Otherwise one node could be elected DC, and later another node be elected DC, and some feature suddenly start or stop working seemingly at random.